pandas/tests/indexes/test_category.py

```diff
@@ -1111,3 +1112,27 @@ def test_take_invalid_kwargs(self):
         msg = "the 'mode' parameter is not supported"
         tm.assert_raises_regex(ValueError, msg, idx.take,
                                indices, mode='clip')
+
+
+class TestCategoricalIndexEngine(object):
+
+    @pytest.mark.parametrize('nbits', [8, 16, 32, 64])
+    def test_engine_type(self, nbits):
```

Comment:
woa, this is very complicated. why is this necessary? can you make much more clear

Reply:
Yeah, I've made it simpler.